src/main/kotlin/com/sourcegraph/cody/edit/sessions/FixupSession.kt

```diff
@@ -95,37 +95,39 @@ abstract class FixupSession(
 
   @RequiresEdt
   private fun triggerFixupAsync() {
-    // Those lookups require us to be on the EDT.
     val file = FileDocumentManager.getInstance().getFile(document)
     val textFile = file?.let { ProtocolTextDocument.fromVirtualFile(editor, it) } ?: return
 
     CodyAgentService.withAgent(project) { agent ->
       workAroundUninitializedCodebase()
 
-      fixupService.startNewSession(this)
-
-      // Spend a turn to get folding ranges before showing lenses.
-      ensureSelectionRange(agent, textFile)
-
-      showWorkingGroup()
-
-      makeEditingRequest(agent)
-          .handle { result, error ->
-            if (error != null || result == null) {
-              showErrorGroup("Error while generating doc string: $error")
-            } else {
-              selectionRange = adjustToDocumentRange(result.selectionRange)
+      try {
+        fixupService.startNewSession(this)
+        // Spend a turn to get folding ranges before showing lenses.
+        ensureSelectionRange(agent, textFile)
+        showWorkingGroup()
+
+        makeEditingRequest(agent)
+            .handle { result, error ->
+              if (error != null) {
+                showErrorGroup("Error while generating code", error)
+              } else if (result == null) {
+                showErrorGroup("Null result generating code for task $taskId")
+              } else {
+                selectionRange = adjustToDocumentRange(result.selectionRange)
+              }
+              null
             }
-            null
-          }
-          .exceptionally { error: Throwable? ->
-            if (!(error is CancellationException || error is CompletionException)) {
-              showErrorGroup("Error while generating code: ${error?.localizedMessage}")
+            .exceptionally { error: Throwable? ->
+              if (!(error is CancellationException || error is CompletionException)) {
+                showErrorGroup("Error while generating code", error)
+              }
+              null
             }
-            cancel()
```

Comment:
Is removal of the manual cancellation intentional?

Reply:
Yes, if we're showing the Error group then dismissing it cancels the session. It's a bit of a stylistic choice to keep the session going until its Error lens is dismissed. One reason is that looking forward to concurrent edits, I'd like to keep error lenses associated with their originating session.

One small user benefit is starting a new session doesn't dismiss your current error group before you've acknowledged or possibly even seen it.